metadata_backend/api/handlers/restapi.py

```diff
@@ -220,6 +220,30 @@ def _header_links(self, url: str, page: int, size: int, total_objects: int) -> C
         LOG.debug("Link headers created")
         return link_headers
 
+    @staticmethod
+    async def iter_submission_objects(
+        submission: dict, obj_op: Operator = None
+    ) -> AsyncIterator[Tuple[str, str, dict]]:
+        """Iterate over a submission's objects.
+
+        :param submission: Submission data
+        :param obj_op: Object Operator
+
+        yields accession_id, schema, object_data
+        """
+        for _obj in submission["metadataObjects"]:
+            accession_id = _obj["accessionId"]
+            schema = _obj["schema"]
+            if obj_op is not None:
+                object_data, _ = await obj_op.read_metadata_object(schema, accession_id)
```

Comment:
we might already have this function as `get_collection_objects` in operators, why not use that ?

I think it replaces this whole iteration ?

Reply:
I'm not sure what do you mean with _accession id filtering_ in this case. What I tried to do was an iterator to go over all objects in a submission. In some cases, all that's needed are the `schema`, in which case `accession_id` and `object_data` are not necessary, and there's no point to read from `mongo`. Then, for the cases when the `object_data` is used, it returns that as well. Could be better if it would be two different methods as well, to make it more clear, and without hidden magic tricks.

Reply:
I see, the maybe all that is required is  what is done at lines 234 -236 i think that is done already by `get_collection_objects`.

What i mean is that we should take advantage of mongo queries when doing this processing.

I don't view mongo queries as hidden tricks, but as means to optimize and speed up the code. There is something else to let the DB give you the formatted result to exactly what you need and we should take advantage of that not read the dict from mongo then process in python. Mongo DB has means to filter and it is optimised for that.

Reply:
I meant as hidden trick lines 237-238 above, that the method is doing something which is not obvious.

Reply:
While that method gets all objects in the same collection, I wanted to iterate over all `metadataObjects` of a submission. So I don't think `get_collection_objects` would fit here.

So I get the impression that explicitly getting each object by ID asynchronously is not so bad. The optimization I'd imagine here would be to use a single `mongo` query, with a cursor, that asynchronously returns each object that belongs to a submission with each iteration. What do you think?

Reply:
sounds ok.